### PR TITLE
Work system

### PR DESCRIPTION
**Pull request by frankitox16**
As it is right now, just shows the current character's name and the job they're doing. Things to do are:

- [x] Add button to cancel job.
- [x] Change job descriptions (Or search for description in localization files).

       ~~Add character sprite in order to visually show the char that's doing the job.~~

- [x] Center the dialog box

preview

### CODE REVIEW

File: Assets/Scripts/UI/DialogBox/Job List/DialogBoxJobList.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    public static readonly Color SecondaryColor = new Color(0.9f, 0.9f, 0.9f);
+
+    public GameObject jobListItemPrefab;
+    public Transform jobList;
+
+    public override void ShowDialog()
+    {
+        base.ShowDialog();
+        
+        foreach (Character c in World.Current.characters)
```
Comment (zwrawr) [inline Assets/Scripts/UI/DialogBox/Job List/DialogBoxJobList.cs]: c should be changed out to a more descriptive name , such as character.

Comment (frankitox16) [inline Assets/Scripts/UI/DialogBox/Job List/DialogBoxJobList.cs]: @zwrawr Given the small amount of use of that var, I decided to keep it that way... As soon as the code involves more usage of that variable, I'll change it, indeed.

**Comment by sboigelot** [comment]
Are you branching out of your local/master instead of the upstream/master?
I used to do the same and had all the past merge in my PR

**Comment by sboigelot** [comment]
(We shouldn't block this PR for that, but i wanted to let you know for your next branch)

**Comment by frankitox16** [comment]
@sboigelot Oh, didn't know that was the way to do it, no wonder why my commit history was so rubbish when I did just 1 commit to the branch itself :no_mouth:

**Comment by sboigelot** [comment]
yep it too me and @vogonistic a while to figure that one out

**Comment by frankitox16** [comment]
@sboigelot thanks for the tip! will do the next time for sure!

**Comment by frankitox16** [comment]
Strings are now searched in localization table, now translators can work with job descriptions! Also centered the dialog box. Quick question, is it really needed the character image?
image

**Comment by TomMalbran** [comment]
This is looking great. Is nice to see a dialog with this information. Small suggestion (can be done later): We could add a double click, or click on the name, on a line that will close the dialog and select the character. Eventually the right selection box will also show the character job.

Someone is also working on the UI for the work button, but is totally different than this one #993. I like to have both dialogs, since both are really useful and do different things, but we need to have 2 different buttons.

**Comment by frankitox16** [comment]
The other dialog box refers more to job priorities, while mine just says current jobs. Maybe a button inside one of the dialog boxes that shows the other one?

**Comment by TomMalbran** [comment]
@frankitox16 Sure things can be added later. After this PR is merged you can make an issue with things that could be improved, if required.

Exactly, one is job priorities and the other current jobs. Until tested properly to see if both are really useful and handy to have at a single test or if one is used more than the other it might be hard to tell if is better to have 2 buttons at the bottom or one open from the other. For now we can leave like this. If the other PR is merged first, you can add a button to the button.

**Comment by sboigelot** [comment]
Testing this now :)

**Comment by sboigelot** [comment]
About the code style:

public field should be replace by public properties:
`public Type MyPropertyName { get; set; }`

You also have a lot of extra blank lines. I can point some to you if you need.

**Comment by sboigelot** [comment]
You should also run StyleCop, there is many warning from your code.

How to run StyleCop: https://github.com/TeamPorcupine/ProjectPorcupine/wiki/StyleCop

**Comment by sboigelot** [comment]
I'll stop here for now as it's still a [WIP], great work so far though!

**Comment by frankitox16** [comment]
Now it should work properly. StyleCop doesn't throw any violations, and everything's quite documented :+1:

**Comment by TomMalbran** [comment]
One small thing. Can you change the folder name from "Job List" to "JobList". It makes things easier when you are in the command line.

**Comment by frankitox16** [comment]
@TomMalbran done :+1:

**Comment by bjubes** [comment]
im getting exceptions when opening the work tab:
```
MissingReferenceException: The variable JobListItemPrefab of DialogBoxJobList doesn't exist anymore.
You probably need to reassign the JobListItemPrefab variable of the 'DialogBoxJobList' script in the inspector.
UnityEngine.Object.Instantiate (UnityEngine.Object original, UnityEngine.Transform parent, Boolean worldPositionStays) (at C:/buildslave/unity/build/Runtime/Export/UnityEngineObject.cs:182)
UnityEngine.Object.Instantiate (UnityEngine.Object original, UnityEngine.Transform parent) (at C:/buildslave/unity/build/Runtime/Export/UnityEngineObject.cs:171)
DialogBoxJobList.Update () (at Assets/Scripts/UI/DialogBox/JobList/DialogBoxJobList.cs:67)
```
The dialog is empty for me as well, probably due to this exception

**Comment by frankitox16** [comment]
@bjubes weird, it works fine by me, and made sure it did before commiting anything... If there's more people with that problem, I'll try recommit the stuff (maybe old meta files? I didn't include them in my commit, but will try next commit then).

**Comment by koosemose** [comment]
No sign of the bug reported by bjubes, if it wasn't something on bjubes' end it was fixed by the last commit. Everything looks good, merging. In the future try to use more descriptive commits than "Yeaah", please.

**Comment by frankitox16** [comment]
@koosemose I will :laughing: I branched out of master and that commit was when I was getting used to git. Next time I'll branch from upstream/master

**Pull request merged**